src/java/org/apache/cassandra/service/accord/db/AccordUpdate.java

```diff
@@ -896,20 +941,46 @@ private static <T> ByteBuffer serialize(T item, IVersionedSerializer<T> serializ
         }
     }
 
-    private static <T> ByteBuffer[] serialize(List<T> items, IVersionedSerializer<T> serializer)
+    private static <T> ByteBuffer[] serialize(Keys keys, List<T> items, Function<? super T, ? extends Key> toKey, IVersionedSerializer<T> serializer)
```

Comment:
A bit odd this is called serialize when it's really populating the arrays and the arrays contain empty byte buffers when the same key appears multiple times.

Reply:
> and the arrays contain empty byte buffers when the same key appears multiple times.

Not sure I follow?

Reply:
`constructValueArrray`?

Reply:
This should be resolved now